ols/src/llms/providers/provider.py

```diff
@@ -44,6 +66,33 @@ def __init__(
         self.model = model
         self.provider_config = provider_config
         self.params = self._override_params(params or {})
+        # self._validate_parameters()
+
+    def _validate_parameters(self) -> None:
+        """Validate parameters for LLM provider."""
+        provider = self.provider_config.type
+
+        if provider is None:
```

Comment:
Can this happen now? If not specified in the config, the name of the provider is used as type.

Reply:
Scratch that. The name can be unset -> None. It is weird but our config allows that...

Reply:
yeah was about to say that ;)